src/QueryM.purs

```diff
@@ -376,11 +469,34 @@ utxoQueryDispatch ref str = do
     let (id :: String) = parsed.reflection.id
     idMap <- Ref.read ref
     let
-      (mAction :: Maybe (JsonWsp.UtxoQR -> Effect Unit)) = (Map.lookup id idMap)
+      (mAction :: Maybe (JsonWsp.UtxoQR -> Effect Unit)) = (MM.lookup id idMap)
     case mAction of
       Nothing -> pure $ (Left (Json.TypeMismatch ("Parse succeeded but Request Id: " <> id <> " has been cancelled")) :: Either Json.JsonDecodeError (Effect Unit))
       Just action -> pure $ Right $ action parsed.result
 
+-- we parse out the utxo query result, then check if we're expecting a result
+-- with the provided id, if we are then we dispatch to the effect that is
+-- waiting on this result
+datumCacheQueryDispatch
+  :: Ref.Ref (MultiMap String (DcWsp.JsonWspResponse -> Effect Unit))
+  -> String
+  -> Effect (Either Json.JsonDecodeError (Effect Unit))
+datumCacheQueryDispatch dim str = do
+  let parsed' = decodeJson =<< Helpers.parseJsonStringifyNumbers str
```

Comment:
Why not merge with @klntsky's datum decoding branch and use the new decoding approach?

Reply:
Will do, since the other pr is very close to being merged I decided to wait couple of hours for it